data-publisher/src/main/kotlin/no/nav/amt/tiltak/data_publisher/publish/ArrangorAnsattPublishQuery.kt

```diff
@@ -0,0 +1,190 @@
+package no.nav.amt.tiltak.data_publisher.publish
+
+import no.nav.amt.tiltak.common.db_utils.DbUtils.sqlParameters
+import no.nav.amt.tiltak.common.db_utils.getLocalDateTime
+import no.nav.amt.tiltak.common.db_utils.getNullableLocalDateTime
+import no.nav.amt.tiltak.common.db_utils.getNullableString
+import no.nav.amt.tiltak.common.db_utils.getUUID
+import no.nav.amt.tiltak.data_publisher.model.*
+import org.springframework.jdbc.core.RowMapper
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.*
+
+class ArrangorAnsattPublishQuery(
+	private val template: NamedParameterJdbcTemplate
+) {
+
+	fun execute(id: UUID): ArrangorAnsattPublishDto {
+		val ansatt = getAnsattDetaljer(id)
+		val roller = getAnsattRoller(id)
+		val veileder = getAnsattVeileder(id)
+		val koordinator = getAnsattKoordinator(id)
+
+		val arrangorIds = roller.map { it.arrangorId }.toSet()
+
+		return ArrangorAnsattPublishDto(
+			id = ansatt.id,
+			personalia = PersonPublishDto(
+				personligIdent = ansatt.personligIdent,
+				navn = Navn(
+					fornavn = ansatt.fornavn,
+					mellomnevn = ansatt.mellomnavn,
+					etternavn = ansatt.etternavn
+				)
+			),
+			arrangorer = arrangorIds.map { arrangorId ->
+				TilknyttetArrangor(
+					arrangorId = arrangorId,
+					roller = roller.filter { it.arrangorId == arrangorId }.map { it.rolle },
+					veileder = veileder.filter { it.arrangorId == arrangorId }.map {
+						Veileder(
+							deltakerId = it.deltakerId,
+							type = if (it.erMedveileder) VeilederType.MEDVEILEDER else VeilederType.VEILEDER
+						)
+					},
+					koordinator = koordinator.filter { it.arrangorId == arrangorId }.map { it.gjennomforingId }
+				)
+			}
+		)
+	}
+
+	private fun getAnsattDetaljer(id: UUID): AnsattDetaljer {
+		return template.query(
+			"SELECT * FROM arrangor_ansatt WHERE id = :id",
+			sqlParameters("id" to id),
+			AnsattDetaljer.rowMapper
+		).first()
+	}
+
+	private fun getAnsattRoller(ansattId: UUID): List<AnsattRoller> {
+		return template.query(
+			"SELECT * FROM arrangor_ansatt_rolle where ansatt_id = :ansattId",
+			sqlParameters("ansattId" to ansattId),
+			AnsattRoller.rowMapper
+		).filter { it.gyldigTil != null && it.gyldigTil.isBefore(LocalDate.of(2500, 1, 1).atStartOfDay()) }
```

Comment:
Hvis gyldigTil er null er vel rollen gyldig og bør sånn sett inkluderes? Hvis vi kun skal ha gyldige roller bør vel it.gyldigTil være etter dagens dato?

Reply:
Samme kommentar gjelder de to andre-gyldig-filtreringene i denne klassen :)

Reply:
ups, takk! Hmm, dette burde det vært en test på!